sphinx_gallery/gen_rst.py

```diff
@@ -236,10 +237,30 @@ def _plots_are_current(src_file, image_file):
     """Test existence of image file and later touch time to source script"""
 
     first_image_file = image_file.format(1)
-    needs_replot = (
-        not os.path.exists(first_image_file) or
-        os.stat(first_image_file).st_mtime <= os.stat(src_file).st_mtime)
-    return not needs_replot
+    has_image = os.path.exists(first_image_file)
+
+    with open(src_file, 'r') as src_data:
+        src_content = src_data.read()
+
+        # data needs to be encoded in python3 before hashing
+        if sys.version_info[0] == 3:
+            src_content = src_content.encode('utf-8')
+
+        src_md5 = hashlib.md5(src_content).hexdigest()
+
+    src_md5_file = src_file + '.md5'
+    src_file_changed = True
+    if os.path.exists(src_md5_file):
+        with open(src_md5_file, 'r') as file_checksum:
+            ref_md5 = file_checksum.read()
+        if src_md5 == ref_md5:
+            src_file_changed = False
+
+    if src_file_changed:
+        with open(src_md5_file, 'w') as file_checksum:
+            file_checksum.write(src_md5)
+
+    return has_image and not src_file_changed
```

Comment:
> That makes me think that actually your md5 hash logic has some flaw. Imagine the example fails for example because one of the depedencies is not installed. You will still write the md5 hash, meaning that once you install the right dependencies the example won't be rerun because its content hasn't changed. Not sure how edge-casy this is and maybe we can live with it for now.

Here is maybe another reason there is the image search